server.js

```diff
@@ -48,24 +47,64 @@ app.get('/', function (req, res) {
 });
 
 app.get('/bunny/:bunnyid', function (req, res, next) {
-  mongoConnection.collection('babies').find({}).toArray(function(err, docs) {
+    mongoConnection.collection('babies').find({}).toArray(function(err, docs) {
     if (err) {
       res.status(500).content('MongoDB failure!');
     }
 
-    if (!(docs[req.params.bunnyid])) {
+    var bun = -1;
+    for(var d in docs){
+        if(docs[d]._id == req.params.bunnyid)
+            bun = d;
+    }
+    if (bun == -1 || !(docs[bun])) {
       next();
       return;
     }
-
     console.log("Found " + docs.length + " babies.");
-    res.status(200).render('index', {
+    res.status(200).render('bunnyDets', {
       title: "Tiffany's Lops",
-      posts: [docs[req.params.bunnyid]]
-    });
+      posts: [docs[bun]]
+     });
   });
 });
 
+app.get('/breeder/', function(req, res, next) {
+    mongoConnection.collection('breeders').find({}).toArray(function(err, docs){
+        if(err) {
+            res.status(500).content('MongoDB failure!');
+	}
+	console.log("Found " + docs.length + " breeders.");
+	res.status(200).render('breeders', {
+	    title: "Tiffany's Lops",
+	    posts: docs
+      	});
+    });
+});
+
+app.get('/breeder/:bunnyid', function(req, res, next) {
+    mongoConnection.collection('breeders').find({}).toArray(function(err, docs){
+        if(err) {
+            res.status(500).content('MongoDB failure!');
+	}
+	var bun = -1;
+	for(var d in docs){
+            if(docs[d]._id == req.params.bunnyid)
+		bun = d;
+	}
+	if(bun == -1 || !(docs[bun])){
+            next();
+	    return;
+	}
+
+	console.log("Found " + docs.length + " breeders");
+        res.status(200).render('breederDets', {
+            title: "Tiffany's Lops",
+	    posts: [docs[bun]]
```

Comment:
If we are using a separate detail page, it should accept only one input, not an array of inputs. This also applies to `/bunny/:bunnyid`.